lib/TransactionKit.ts

```diff
@@ -864,861 +1566,2305 @@ export class EtherspotTransactionKit implements IInitial {
+          // MULTI-CHAIN GROUPING: Separate transactions by chainId
+          // ====================================================================
+
+          // Group transactions by chainId for separate estimation per chain
+          const chainIdToTxs = new Map<number, typeof batchTransactions>();
+          for (const tx of batchTransactions) {
+            const txChainId =
+              tx.chainId ?? this.#etherspotProvider.getChainId();
+            const list = chainIdToTxs.get(txChainId) || [];
+            list.push(tx);
+            chainIdToTxs.set(txChainId, list);
+          }
+
+          // ====================================================================
+          // CHAIN GROUP ESTIMATION: Process each chain group sequentially
+          // ====================================================================
+          for (const [groupChainId, groupTxs] of chainIdToTxs.entries()) {
+            const groupEstimated: TransactionEstimateResult[] = [];
+            try {
+              log(
+                `estimateBatches(): Getting delegatedEoa account and bundler client for batch ${batchName} on chain ${groupChainId}...`,
+                undefined,
+                this.debugMode
+              );
+              const delegatedEoaAccount =
+                await this.#etherspotProvider.getDelegatedEoaAccount(
+                  groupChainId
+                );
+              const bundlerClient =
+                await this.#etherspotProvider.getBundlerClient(groupChainId);
+
+              log(
+                `estimateBatches(): Got account ${delegatedEoaAccount.address} and bundler client for batch ${batchName}`,
+                undefined,
+                this.debugMode
+              );
+
+              // ====================================================================
+              // CALL PREPARATION: Convert transactions to viem call format
+              // ====================================================================
+
+              // Prepare calls for this chain group
+              const calls = groupTxs.map((tx) => ({
+                to: (tx.to || '') as `0x${string}`,
+                value: BigInt(tx.value?.toString() || '0'),
+                data: (tx.data || '0x') as `0x${string}`,
+              }));
+
+              log(
+                `estimateBatches(): Prepared ${calls.length} calls for batch ${batchName} (chain ${groupChainId})`,
+                calls,
+                this.debugMode
+              );
+
+              // ====================================================================
+              // EIP-7702 VALIDATION: Check if EOA is designated for smart wallet
+              // ====================================================================
+
+              // Ensure EOA is designated (EIP-7702) on this chain
+              const isDesignated =
+                await this.isDelegateSmartAccountToEoa(groupChainId);
+              if (!isDesignated) {
```

Comment:
unless i missed something there and based on the next line also - if the EOA is not delegated should we not fail fast? It looks like we're continuing here